UnityGameBase/Core/Utils/Extensions.cs

```diff
@@ -89,7 +89,12 @@ public static T GetRandomAndRemove<T>(this List<T> target)
         
         public static void Shuffle<T>(this List<T> list)
         {
-            System.Random rng = new System.Random();  
+            // System.Random() initialisation is obviously based on a low
+            // resolution timestamp of the current time, which yields
+            // identical shuffles if called in close succession.
+            // Solution: Use milliseconds, which provide more variation.
+            System.Random rng = new System.Random(System.DateTime.Now.Millisecond);
```

Comment:
#8 looks fine, I have not noticed it before. It should throw a warning due to the implicit downcast, though, or maybe it doesn't?

Regarding the comments: as mentioned above, the resolution of the default implementation is indeed low (10 to 16 ms on desktop machines) that's why I was merely suggesting to enhance it instead of completely replacing it. Using the millisecond part of the current date would have fixed the problem for short-term evaluations but it would keep generating the same numbers in the long run -- which is what I was concerned about.

Out of curiosity, why would it *not* be "the most reasonable" idea to make it (private) static? It's merely for caching the RNG instance. Multithreading is not an issue here so I must be missing something else?